functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+func getPodLogs(pod v1core.Pod) (string, error) {
+	podLogOpts := v1core.PodLogOptions{}
+	req := client.Client.CoreV1Interface.Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+
+	return str, nil
+}
+
+func netParseEthtoolTimeStampFeature(cmdOut *bytes.Buffer) bool {
+	var hardRxEnabled bool
+	var hardTxEnabled bool
+	var hardRawEnabled bool
+
+	glog.V(2).Infof("cmd output for %v", cmdOut)
+	scanner := bufio.NewScanner(cmdOut)
+	for scanner.Scan() {
+		line := strings.TrimPrefix(scanner.Text(), "\t")
+		parts := strings.Fields(line)
+		if parts[0] == _ETHTOOL_HARDWARE_RECEIVE_CAP {
+			hardRxEnabled = parts[1] == _ETHTOOL_RX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_TRANSMIT_CAP {
+			hardTxEnabled = parts[1] == _ETHTOOL_TX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_RAW_CLOCK_CAP {
+			hardRawEnabled = parts[1] == _ETHTOOL_RAW_HARDWARE_FLAG
+		}
+	}
+	return hardRxEnabled && hardTxEnabled && hardRawEnabled
+}
+
+func ptpDisoveredInterfaceList(path string) []string {
+	var ptpInterfaces []string
+	fg, err := client.Client.CoreV1Interface.RESTClient().Get().AbsPath(path).DoRaw()
+	Expect(err).ToNot(HaveOccurred())
+	var nodePtpDevice ptpv1.NodePtpDevice
+	err = json.Unmarshal(fg, &nodePtpDevice)
+	Expect(err).ToNot(HaveOccurred())
+	for _, intConf := range nodePtpDevice.Status.Devices {
+		ptpInterfaces = append(ptpInterfaces, intConf.Name)
+	}
+	return ptpInterfaces
+}
+
+func detectDefaultRouteInterface(routeTable string) (string, error) {
+	for _, route := range strings.Split(routeTable, "\n")[1 : len(strings.Split(routeTable, "\n"))-1] {
+		if strings.Split(strings.Join(strings.Fields(route), " "), " ")[1] == "00000000" &&
+			strings.Split(strings.Join(strings.Fields(route), " "), " ")[7] == "00000000" {
+			return strings.Split(strings.Join(strings.Fields(route), " "), " ")[0], nil
+		}
+	}
+	return "", fmt.Errorf("Default route not present")
+}
+
+func mutateProfile(profile ptpv1.PtpConfig, profileName string, nodeName string) *ptpv1.PtpConfig {
+	mutatedConfig := profile.DeepCopy()
+	priority := int64(100)
+	mutatedConfig.ObjectMeta.Reset()
+	mutatedConfig.ObjectMeta.Name = "test"
+	mutatedConfig.ObjectMeta.Namespace = ptpLinuxDaemonNamespace
+	mutatedConfig.Spec.Profile[0].Name = &profileName
+	mutatedConfig.Spec.Recommend[0].Priority = &priority
+	mutatedConfig.Spec.Recommend[0].Match[0].NodeLabel = nil
+	mutatedConfig.Spec.Recommend[0].Match[0].NodeName = &nodeName
+	mutatedConfig.Spec.Recommend[0].Profile = &profileName
+	return mutatedConfig
+}
+
+func waitUntilDsPodsReady(namespace string, DaemonSetName string, timeout int) error {
+	for timeout > 0 {
```

Comment:
please use here the `eventually` function in ginkgo

Reply:
No need in this function. I remove it